main.go

```diff
@@ -170,3 +209,78 @@ const testFileTemplate = `
 func formatTestFile(method string, service string) string {
 	return fmt.Sprintf(testFileTemplate, method, service, method)
 }
+
+// add in reflection api
+const serverTemplate = `
+func main() {
+    if err := run(); err != nil {
+        log.Fatal(err)
+    }
+}
+
+func run() error {
+    listenOn := "127.0.0.1:8080"
+    listener, err := net.Listen("tcp", listenOn)
+    if err != nil {
+        return  err 
+    }
+
+    server := grpc.NewServer()
+    %s.Register%sServer(server, &%s{}) // this would need to be a list or multiple.
+	reflection.Register(server) // this should perhaps be optional
+    log.Println("Listening on", listenOn)
+    if err := server.Serve(listener); err != nil {
+        return err 
+    }
+
+    return nil
+}
+
+`
+
+// need pkg, services,
+func generateServer(gen *protogen.Plugin, file *protogen.File) {
+	services := []string{}
+
+	for _, v := range file.Services {
+		services = append(services, v.GoName) // service.Desc.Name()
+	}
+
+	fileName := strings.ToLower("cmd" + "/" + string(file.GoPackageName) + "/" + "main.go")
+	g := gen.NewGeneratedFile(fileName, protogen.GoImportPath("."))
+
+	g.P("package main ")
+
+	// required imports
+	g.QualifiedGoIdent(protogen.GoIdent{GoImportPath: file.GoImportPath})
+	g.QualifiedGoIdent(protogen.GoIdent{GoImportPath: "log"})
+	g.QualifiedGoIdent(protogen.GoIdent{GoImportPath: "net"})
+	g.QualifiedGoIdent(protogen.GoIdent{GoImportPath: "google.golang.org/grpc"})
+	g.QualifiedGoIdent(protogen.GoIdent{GoImportPath: "google.golang.org/grpc/reflection"})
+
+	// need to be for loop , hardcoding for now
+	serviceName := services[0]
+
+	// this is slightly flawed as if someone uses a repeated val in their proto path
+	//  the import could end up pretty cooked.
+	protoName := strings.Trim(*file.Proto.Name, ".proto")
+	goModPath := strings.Replace(file.GeneratedFilenamePrefix, string(file.GoPackageName), "", 1)
+	goModPath = strings.Replace(goModPath, string(protoName), "", 1)
+	goModPath = strings.Trim(goModPath, "/")
```

Comment:
this probably wont work for proto files that are not in the same directory. 

options i can think of now are 
- pass in gomod path to plugin
- 2 plugins (one for service gen 1 for server)